master/internal/api_master.go

```diff
@@ -25,3 +26,60 @@ func (a *apiServer) GetMaster(
 		Config:    configStruct,
 	}, err
 }
+
+// effectiveOffset Returns effective offset.
+func effectiveOffset(reqOffset int, totalItems int) (offset int) {
+	offset = reqOffset
```

Comment:
reqOffset comes in as a copy so it should be safe to override the input but what's the best practice for Go, do we override the input?

Reply:
Yeah it's totally safe, but think altering params can be confusing still. I think something like this is much clearer, as far as the intent of what we're doing and that we don't even get into this issue:
```
func effectiveOffset(offset int, total int) int {
	switch {
        case offset < -total:
                return 0
        case offset < 0:
                return total + offset
        default:
                return offset
	}
}
```